File: CSharpStuff/Nathandelane.Math.Processor/Nathandelane.Math.Processor.Tokens/Arithmetic/Multiplication.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Nathandelane.Math.Processor.Tokens
{
	public class Multiplication : IToken
	{
		#region Fields

		private TokenType _type;
		private TokenPrecedence _precedence;
		private string _value;

		#endregion

		#region Constructors

		public Multiplication()
		{
			_type = TokenType.Operator;
			_precedence = new TokenPrecedence(3);
			_value = "*";
		}

		#endregion

		#region IToken Members

		public TokenType Type
		{
			get { return _type; }
		}

		public TokenPrecedence Precedence
		{
			get
			{
				return _precedence;
			}
			set
			{
				_precedence = value;
			}
		}

		public string Value
		{
			get { return _value; }
		}

		#endregion
	}
}

File: JavaStuff/PersonalCalculator/personalcalculator/src/com/nathandelane/math/evaluation/tokens/Multiplication.java
package com.nathandelane.math.evaluation.tokens;

import java.util.regex.Pattern;

public class Multiplication extends TokenBase {

	public Multiplication() {
		super(TokenType.OPERATOR, TokenPrecedence.MULTIPLICATION_DIVISION, "*");
		
		_pattern = Pattern.compile("^[*]{1}");
	}
	
}
